Mathlib/NumberTheory/EulerProduct/Basic.lean

```diff
@@ -146,11 +266,61 @@ section CompletelyMultiplicative
 
 We now assume that `f` is completely multiplicative and has values in a complete normed field `F`.
 Then we can use the formula for geometric series to simplify the statement. This leads to
-`EulerProduct.eulerProduct_completely_multiplicative`.
+`EulerProduct.eulerProduct_completely_multiplicative_hasProd` and variants.
 -/
 
 variable {F : Type*} [NormedField F] [CompleteSpace F]
 
+-- local instances to speed up typeclass search
+@[nolint defLemma docBlame] local instance : TopologicalSpace.MetrizableSpace F :=
+  MetricSpace.toMetrizableSpace
+@[nolint defLemma docBlame] local instance : T2Space F :=
+  TopologicalSpace.t2Space_of_metrizableSpace
+@[nolint defLemma docBlame] local instance : One F := Semiring.toOne
```

Comment:
This only saves 3 heartbeats over `inferInstance`. Of these four custom instances, certainly the first three don't seem to give a significant speedup (none of them take more than 0.05sec on my laptop). For the fourth one, the automation is a little bit slower but it's still only 0.13 sec. 

This feeds into the debate we were having on zulip not long back : how much slowness are we willing to tolerate in exchange for the convenience of automation? 

(Unrelated: I wonder why the `defLemma` linter is firing on these?)